lib/linguist/generated.rb

```diff
@@ -627,6 +628,16 @@ def generated_visualstudio6?
       lines.first(3).any? { |l| l.include? '# Microsoft Developer Studio Generated Build File' }
     end
 
+    HAXE_EXTENSIONS = ['.js', '.py', '.lua', '.cpp', '.h', '.java', '.cs', '.php']
```

Comment:
I've seen this pattern of checking the file extension in other `generated`-functions. I'm assuming this is desired for performance reasons? Otherwise, not checking the extension at all would be a bit easier to maintain (Haxe adds new targets from time to time).

Reply:
> I'm assuming this is desired for performance reasons?

Yes, we want to avoid reading the first three lines of every single file on GitHub.

Reply:
Ok, makes sense. :+1: